pkg/register/restResolver.go

```diff
@@ -76,10 +76,19 @@ func (c *RestResolverClient) GetDocument(ctx context.Context, documentID string)
 		return nil, err
 	}
 
-	discoverURL := fmt.Sprintf("%s/1.0/discover/%s", c.url.String(), url.QueryEscape(documentID)) // todo: join path?
-	response, err := c.client.Get(discoverURL)
+	discoverURL := fmt.Sprintf("%s/1.0/discover/%s", c.url.String(), url.QueryEscape(documentID))
 
+	reqWithContext, err := http.NewRequestWithContext(ctx, http.MethodGet, discoverURL, nil)
 	if err != nil {
+		return nil, &ResolverError{err: err, errType: ServerError}
+	}
+	response, err := c.client.Do(reqWithContext)
```

Comment:
One more thing (only just realised): `response.Body` should be defer-closed in the non-error case